sdk/clientcore/core/src/main/java/io/clientcore/core/util/auth/ChallengeHandler.java

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.util.auth;
+
+import io.clientcore.core.http.models.HttpRequest;
+import io.clientcore.core.http.models.Response;
+import io.clientcore.core.implementation.util.auth.DigestChallengeHandler;
+import io.clientcore.core.implementation.util.auth.DigestProxyChallengeHandler;
+import io.clientcore.core.util.ClientLogger;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Class representing a challenge handler for authentication.
+ */
+public interface ChallengeHandler {
+    ClientLogger LOGGER = new ClientLogger(ChallengeHandler.class);
+
+    /**
+     * Handles the authentication challenge based on the HTTP request and response.
+     * @param request The HTTP request to be updated with authentication info.
+     * @param response The HTTP response containing the authentication challenge.
+     */
+    void handleChallenge(HttpRequest request, Response<?> response);
```

Comment:
After reviewing more of the PR, I think in both APIs here we should add one more parameter, something along the lines of `boolean isProxy`. This would allow us to remove the duplicative `DigestProxyChallengeHandler` and have just `DigestChallengeHandler`. And should let us remove the code in the OkHttp client except the `Authenticator` implementations (it'd just be a high-level passthrough to the ChallengeHandler being used).